src/mbgl/util/tile_range.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <mbgl/tile/tile_id.hpp>
+#include <mbgl/util/range.hpp>
+#include <mbgl/util/geo.hpp>
+#include <mbgl/util/projection.hpp>
+
+namespace mbgl {
+
+namespace util {
+
+class TileRange {
+public:
+    Range<Point<double>> range;
+    uint8_t z;
+
+    // Compute the range of tiles covered by the bounds.
+    static TileRange fromLatLngBounds(const LatLngBounds& bounds, uint8_t z) {
+        auto swProj = Projection::project(bounds.southwest().wrapped(), z);
+        auto ne = bounds.northeast();
+        auto neProj = Projection::project(ne.longitude() > util::LONGITUDE_MAX ? ne.wrapped() : ne , z);
+        auto minX = std::floor(swProj.x);
+        auto maxX = std::ceil(neProj.x);
+        auto minY = std::floor(neProj.y);
+        auto maxY = std::ceil(swProj.y);
+        return TileRange({ {minX, minY}, {maxX, maxY} }, z);
+    }
+
+    bool contains(const CanonicalTileID& tileID) {
+        return z == tileID.z &&
+        (range.min.x >= range.max.x ? //For wrapped bounds
```

Comment:
fwiw I don't think wrapped bounds are supported in TileJSON – I'm not sure if wrapped (i.e. `> 180 || < -180`) latitudes are considered to be true WSG:84 coordinates but 🤷‍♀️ 

```
    // OPTIONAL. Default: [-180, -90, 180, 90].
    // The maximum extent of available map tiles. Bounds MUST define an area
    // covered by all zoom levels. The bounds are represented in WGS:84
    // latitude and longitude values, in the order left, bottom, right, top.
    // Values may be integers or floating point numbers.
    "bounds": [ -180, -85.05112877980659, 180, 85.0511287798066 ],
```
https://github.com/mapbox/tilejson-spec/tree/master/2.2.0

Reply:
in the JS implementation we also clamp the bounds, so this would introduce differing behavior between the two packages
https://github.com/mapbox/mapbox-gl-js/blob/2124161ebba0c17d764e2e7f78e2d34a283e9bae/src/source/tile_bounds.js#L22

Reply:
The input coordinates can easily be clamped, but if left> right, it will result in the unwrapped bounds in native, and invalid bounds in JS, where the `contains` test will always fail on longitude [here](https://github.com/mapbox/mapbox-gl-js/blob/2124161ebba0c17d764e2e7f78e2d34a283e9bae/src/source/tile_bounds.js#L32).